src/main/java/gregtech/api/util/GT_RecipeBuilder.java

```diff
@@ -48,12 +55,18 @@ public class GT_RecipeBuilder {
     static {
         if (System.getProperties()
             .containsKey("gt.recipebuilder.debug")) {
-            DEBUG_MODE = Boolean.getBoolean("gt.recipebuilder.debug");
+            DEBUG_MODE_ALL = Boolean.getBoolean("gt.recipebuilder.debug");
         } else {
             // turn on debug by default in dev mode
-            DEBUG_MODE = (Boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment");
+            DEBUG_MODE_ALL = (boolean) Launch.blackboard.get("fml.deobfuscatedEnvironment");
         }
-        PANIC_MODE = DEBUG_MODE && Boolean.getBoolean("gt.recipebuilder.panic");
+        DEBUG_MODE_NULL = Boolean.getBoolean("gt.recipebuilder.debug.null");
+        DEBUG_MODE_INVALID = Boolean.getBoolean("gt.recipebuilder.debug.invalid");
+        DEBUG_MODE_COLLISION = Boolean.getBoolean("gt.recipebuilder.debug.collision");
+        PANIC_MODE_ALL = Boolean.getBoolean("gt.recipebuilder.panic");
+        PANIC_MODE_NULL = Boolean.getBoolean("gt.recipebuilder.panic.null");
+        PANIC_MODE_INVALID = Boolean.getBoolean("gt.recipebuilder.panic.invalid");
+        PANIC_MODE_COLLISION = Boolean.getBoolean("gt.recipebuilder.panic.collision");
```

Comment:
Maybe we should enable specific debug/panic levels when global level is enabled. This change will eliminate need to check `ALL` flags every time.
```suggestion
        DEBUG_MODE_NULL = DEBUG_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.debug.null");
        DEBUG_MODE_INVALID = DEBUG_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.debug.invalid");
        DEBUG_MODE_COLLISION = DEBUG_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.debug.collision");
        PANIC_MODE_ALL = Boolean.getBoolean("gt.recipebuilder.panic");
        PANIC_MODE_NULL = PANIC_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.panic.null");
        PANIC_MODE_INVALID = PANIC_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.panic.invalid");
        PANIC_MODE_COLLISION = PANIC_MODE_ALL || Boolean.getBoolean("gt.recipebuilder.panic.collision");
```